server/service/osquery_utils/queries.go

```diff
@@ -164,6 +158,64 @@ var hostDetailQueries = map[string]DetailQuery{
 			return nil
 		},
 	},
+	"os_version_unix_like": {
+		// Collect platform-specific information to populate `host.OSVersion` for Darwin and Linux.
+		// Note that data for `operating_system` and `host_operating_system` tables are ingested via
+		// the `os_unix_like` extra detail query below.
+		Query:     `SELECT * FROM os_version LIMIT 1`,
```

Comment:
I opted for this approach rather because in order to populate the `os_version` column we need to run a Windows-specific registry query that fails on non-Windows hosts. So it made sense to me do the full select once to just populate the platform, then have two distinct queries to populate the rest of the info for the Windows vs. Unix-like hosts. We could cut out the extra Unix-like query with a conditional in the first query if you prefer that approach.

Reply:
Yes, I get the `os_version_windows` one.

But unless am missing something both `os_version` and `os_version_unix_like` run `SELECT * FROM os_version LIMIT 1`.

- `os_version` runs on all hosts.
- `os_version_unix_like` runs on darwin+linux hosts.

So for e.g. a darwin host, `SELECT * FROM os_version LIMIT 1` is sent twice.

My proposal was maybe we could remove `os_version_unix_like` and have something like:
```go
	"os_version": {
		// This query is required to collect basic platform information to enroll new hosts.
		// Once we know the platform, we will we can make platform-specific queries to populate
		// `host.OSVersion` via `os_version_windows` and `os_version_unix_like` below
		Query: "SELECT * FROM os_version LIMIT 1",
		IngestFunc: func(ctx context.Context, logger log.Logger, host *fleet.Host, rows []map[string]string) error {
			if len(rows) != 1 {
				logger.Log("component", "service", "method", "IngestFunc", "err",
					fmt.Sprintf("detail_query_os_version expected single result got %d", len(rows)))
				return nil
			}

			if build, ok := rows[0]["build"]; ok {
				host.Build = build
			}

			host.Platform = rows[0]["platform"]
			host.PlatformLike = rows[0]["platform_like"]
			host.CodeName = rows[0]["codename"]

			// On centos6 there is an osquery bug that leaves
			// platform empty. Here we workaround.
			if host.Platform == "" &&
				strings.Contains(strings.ToLower(rows[0]["name"]), "centos") {
				host.Platform = "centos"
			}

                         ///////////////////////////// extra code for UNIX here /////////////////////////////
                         if fleet.IsUnix(host) {
                                 host.OSVersion = fmt.Sprintf("%v %v", rows[0]["name"], parseOSVersion(
				    rows[0]["name"],
				    rows[0]["version"],
				    rows[0]["major"],
				    rows[0]["minor"],
				    rows[0]["patch"],
				    rows[0]["build"],
			))
                         }

			return nil
		},
	},
```

Reply:
Thanks! I was working on something similar as I was addressing some merge conflicts :)